packages/commands/src/query/cctx.ts

```diff
@@ -0,0 +1,253 @@
+import axios from "axios";
+import { Command } from "commander";
+import { ethers } from "ethers";
+import EventEmitter from "eventemitter3";
+import { z } from "zod";
+
+import { CrossChainTx } from "../../../../types/cctx";
+
+/**
+ * Event map:
+ *  - `cctx` → emitted every polling round with the **entire** array so far.
+ */
+interface CctxEvents {
+  cctx: (allSoFar: CrossChainTx[]) => void;
+}
+
+export const cctxEmitter = new EventEmitter<CctxEvents>();
+
+const cctxOptionsSchema = z.object({
+  delay: z.coerce.number().int().positive().default(2000),
+  hash: z.string(),
+  rpc: z.string(),
+});
+
+type CctxOptions = z.infer<typeof cctxOptionsSchema>;
+
+interface CctxResponse {
+  CrossChainTxs: CrossChainTx[];
+}
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const fetchCctx = async (
+  hash: string,
+  rpc: string
+): Promise<CrossChainTx[]> => {
+  const url = `${rpc}/zeta-chain/crosschain/inboundHashToCctxData/${hash}`;
+  const res = await axios.get<CctxResponse>(url, {
```

Comment:
We could reuse the `fetchFromApi` helper here

Reply:
93b5278 had to remove console logging from `catch`.